lib/src/headers.dart

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:collection/collection.dart';
+
+import 'http_unmodifiable_map.dart';
+
+/// Adds a header with [name] and [value] to [headers], which may be null.
+///
+/// Returns a new map without modifying [headers].
+Map<String, String> addHeader(
```

Comment:
Just `getHeader` is used so far. I wasn't completely positive if the others would be of use as we moved further. I'll remove and add back if needed.